src/playground-code-editor.ts

```diff
@@ -181,6 +182,30 @@ export class PlaygroundCodeEditor extends LitElement {
     this.requestUpdate('value', oldValue);
   }
 
+  /**
+   * Fallback documentKey which is used for the default document instance.
+   */
+  private readonly _fallbackDocInstance = {};
+
+  /**
+   * Provide a `documentKey` to create a CodeMirror document instance which
+   * isolates history and value changes per `documentKey`.
+   *
+   * Use to keep edit history separate between files while reusing the same
+   * playground-code-editor instance.
+   */
+  @property({attribute: false})
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  documentKey?: object = this._fallbackDocInstance;
```

Comment:
`this._fallbackDocInstance` being the default here means that the special key can be read by somebody outside, and it also might be unintuitive that it would not be `undefined` when the user hasn't set it yet. Seems like we should probably keep the fallback key completely private?

Reply:
Your excellent suggestion in the other comment let me remove this idea. Also that is a great point that I hadn't considered.